app/routes/organen/orgaan/mandatarissen.js

```diff
@@ -88,80 +89,105 @@ export default class OrganenMandatarissenRoute extends Route {
       })
     );
-    return folded.map((entry) => {
-      const fracties = entry.foldedFracties;
-      fracties.sort((a, b) => a.localeCompare(b));
-      const currentFractie = entry.fractie;
-      const otherFracties = fracties.filter((f) => f != entry.fractie && f);
-      const fractieText = otherFracties.length
-        ? `${currentFractie} (${otherFracties.join(', ')})`
-        : currentFractie;
-      return {
-        mandataris: entry.mandataris,
-        foldedStart: entry.foldedStart,
-        foldedEnd: entry.foldedEnd,
-        foldedFracties: fractieText,
-      };
-    });
+    return Object.values(persoonMandaatData).map(
+      ({ foldedStart, foldedEnd, mandataris, fractie, foldedFracties }) => {
+        return {
+          foldedStart,
+          foldedEnd,
+          mandataris,
+          foldedFracties: this.fractiesToString(fractie, foldedFracties),
+        };
+      }
+    );
+  }
+
+  updateFoldedMandataris(mandataris, fractie, foldedMandataris) {
+    this.updateFoldedStart(mandataris, foldedMandataris);
+    this.updateFoldedEnd(mandataris, foldedMandataris);
+    this.updateFoldedFracties(fractie, foldedMandataris);
+    this.updateFoldedMandatarisAndFractie(
+      mandataris,
+      fractie,
+      foldedMandataris
+    );
+  }
+
+  updateFoldedStart(mandataris, foldedMandataris) {
+    if (!mandataris.start || !foldedMandataris.foldedStart) {
+      foldedMandataris.foldedStart = null;
+    } else {
+      foldedMandataris.foldedStart = moment.min(
+        moment(foldedMandataris.foldedStart),
+        moment(mandataris.start)
+      );
+    }
+  }
+
+  updateFoldedEnd(mandataris, foldedMandataris) {
+    if (!mandataris.einde || !foldedMandataris.foldedEnd) {
+      foldedMandataris.foldedEnd = null;
+    } else {
+      foldedMandataris.foldedEnd = moment.max(
+        moment(foldedMandataris.foldedEnd),
+        moment(mandataris.einde)
+      );
+    }
+  }
+
+  updateFoldedFracties(fractie, foldedMandataris) {
+    if (fractie && !foldedMandataris.foldedFracties.includes(fractie)) {
+      foldedMandataris.foldedFracties.push(fractie);
+    }
+  }
+
+  updateFoldedMandatarisAndFractie(mandataris, fractie, foldedMandataris) {
+    // keep the one with the latest end date (if it is null, we assume this is the latest one)
+    if (
+      moment(mandataris.einde).isSame(foldedMandataris.foldedEnd) ||
+      !mandataris.einde
+    ) {
+      foldedMandataris.mandataris = mandataris;
+      foldedMandataris.fractie = fractie;
+    }
+  }
+
+  buildFoldedMandataris(mandataris, fractie) {
+    const fracties = fractie ? [fractie] : [];
+
+    return {
+      foldedStart: mandataris.start,
+      foldedEnd: mandataris.einde,
+      mandataris,
+      fractie,
+      foldedFracties: fracties,
+    };
+  }
+
+  fractiesToString(currentFractie, allFracties) {
+    const sortedFracties = allFracties
+      .filter((f) => f != currentFractie && f) // TODO the "&& f" is confusing
```

Comment:
Inherited from routes/organen/orgaan/mandatarissen foldMandatarissen method, but not sure why the filter works this way